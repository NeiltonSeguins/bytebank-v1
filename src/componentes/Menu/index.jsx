import React from 'react';
import estilos from './Menu.module.css';

const listaMenu = [
  { link: 'Inicial', href: '/' },
  { link: 'Cartões', href: '/cartoes' },
  { link: 'Transferências', href: '/transferencias' },
  { link: 'Investimentos', href: '/investimentos' },
  { link: 'Outros serviços', href: '/outros-servicos' },
];

export default function Menu() {
  return (
    <nav className={estilos.menu}>
      {listaMenu.map((item, indice) => {
        return (
          <div key={item.href} className={estilos.item}>
            <a href={item.href} className={estilos.link}>
              {item.link}
            </a>
            {indice !== listaMenu.length - 1 && (
              <div className={estilos.divisor} />
            )}
          </div>
        );
      })}
    </nav>
  );
}
