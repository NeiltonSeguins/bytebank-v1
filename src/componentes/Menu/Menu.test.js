import { render, screen } from '@testing-library/react';
import Menu from './index';

test('Deve renderizar o link para a página inicial', () => {
  render(<Menu />);
  const linkPaginaInicial = screen.getByText('Inicial');
  expect(linkPaginaInicial).toBeInTheDocument();
});

test('Deve renderizar uma lista com quatro links', () => {
  render(<Menu />);
  const linksPaginaInicial = screen.getAllByRole('link');
  expect(linksPaginaInicial).toHaveLength(4);
});

test('Não deve renderizar o link de Extrato', () => {
  render(<Menu />);
  const linkExtrato = screen.queryByText('Extrato');
  expect(linkExtrato).not.toBeInTheDocument();
});
