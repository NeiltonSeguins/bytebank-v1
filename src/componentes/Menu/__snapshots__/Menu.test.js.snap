// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deve renderizar os links com a classe link 1`] = `
Array [
  <a
    class="link"
    href="/"
  >
    Inicial
  </a>,
  <a
    class="link"
    href="/cartoes"
  >
    Cartões
  </a>,
  <a
    class="link"
    href="/transferencias"
  >
    Transferências
  </a>,
  <a
    class="link"
    href="/investimentos"
  >
    Investimentos
  </a>,
  <a
    class="link"
    href="/outros-servicos"
  >
    Outros serviços
  </a>,
]
`;
